lua/starfall/libs_sh/players.lua

```diff
@@ -618,6 +668,116 @@ if SERVER then
 		checkluatype(val, TYPE_NUMBER)
 		ent:SetMaxArmor(val)
 	end
+	
+	--- Sets Crouched Walk Speed
+	-- @server
+	-- @param number newcwalkspeed New Crouch Walk speed, This is a multiplier from 0 to 1.
+	function player_methods:setCrouchedWalkSpeed(val)
+		local ent = getply(self)
+		checkpermission(instance, ent, "player.modifyMovementProperties")
+		checkluatype(val, TYPE_NUMBER)
+		ent:SetCrouchedWalkSpeed(math.Clamp(val,0,1))
+	end
+	
+	--- Sets Duck Speed
+	-- @server
+	-- @param number newduckspeed New Duck speed, This is a multiplier from 0 to 1.
+	function player_methods:setDuckSpeed(val)
+		local ent = getply(self)
+		checkpermission(instance, ent, "player.modifyMovementProperties")
+		checkluatype(val, TYPE_NUMBER)
+		ent:SetDuckSpeed(math.Clamp(val,0.005,0.995))
```

Comment:
Wouldn't a value greater than 1 also be wanted in some cases? I'd say 0.001 to 1000 would be acceptable clamping range.

Reply:
https://github.com/Facepunch/garrysmod-issues/issues/2722